uk.co.compucorp.civicrm.hrleaveandabsences/gulp-tasks/requirejs.js

```diff
@@ -0,0 +1,41 @@
+var gulp = require('gulp');
+var clean = require('gulp-clean');
+var rename = require('gulp-rename');
+var path = require('path');
+
+module.exports = function () {
+  // @NOTE one thing to keep in mind is that the custom task always needs to use
+  // __dirname for paths, otherwise the paths will be relative to hrcore
+  var distFolder = path.join(__dirname, '..', 'js/angular/dist/');
+
+  return {
+    // @NOTE here L&A needs to perform to tasks after r.js had run, both for
+    // cleaning the dist/ folder. Notice that the order of the tasks is the
+    // order in which they will be called
+    post: [
```

Comment:
Can we further split the post and pre tasks into different files?

```js
// gulp-tasks/requirejs.js
return {
  post: [
    require('./post-requirejs-rename'),
    require('./post-requirejs-clean')
  ]
}
```

Or, the main gulpfile can have a few extra hooks: requirejs-rename and requirejs-clean.

Reply:
not sure what you mean about the extra hooks, the whole idea is that the main gulpfile doesn't know and doesn't need to know about whatever custom logic a single extension needs to run